pkg/core/resources/apis/meshservice/hostname/generator_test.go

```diff
@@ -143,19 +152,114 @@ var _ = Describe("MeshService Hostname Generator", func() {
 		}, "2s", "100ms").Should(Succeed())
 	})
 
+	It("should generate hostnames for services in a consistent order", func() {
+		backend := samples.MeshServiceBackendBuilder().WithoutVIP().WithLabels(map[string]string{
+			"generate": "static",
+		})
+		other := samples.MeshServiceBackendBuilder().WithoutVIP().WithLabels(map[string]string{
+			"generate": "static",
+		}).WithName("other")
+		createdFirst := time.Now()
+		createdSecond := createdFirst
+		createdSecond = createdSecond.Add(1 * time.Minute)
+
+		// when
+		Expect(backend.Create(resManager, core_store.CreatedAt(createdFirst))).To(Succeed())
+		Expect(other.Create(resManager, core_store.CreatedAt(createdSecond))).To(Succeed())
+
+		Expect(builders.HostnameGenerator().
+			WithName("static").
+			WithTemplate("static.mesh").
+			WithMeshServiceMatchLabels(map[string]string{"generate": "static"}).
+			Create(resManager),
+		).To(Succeed())
+
+		// then
+		Eventually(func(g Gomega) {
+			otherStatus := meshServiceStatus("other")
+			backendStatus := meshServiceStatus("backend")
+			g.Expect(otherStatus.Addresses).Should(BeEmpty())
+			g.Expect(otherStatus.HostnameGenerators).Should(ConsistOf(
+				hostnamegenerator_api.HostnameGeneratorStatus{
+					HostnameGeneratorRef: hostnamegenerator_api.HostnameGeneratorRef{CoreName: "static"},
+					Conditions: []hostnamegenerator_api.Condition{{
+						Type:    hostnamegenerator_api.GeneratedCondition,
+						Status:  kube_meta.ConditionFalse,
+						Reason:  hostnamegenerator_api.CollisionReason,
+						Message: "Hostname collision with MeshService: other",
+					}},
+				},
+			))
+			g.Expect(backendStatus.Addresses).Should(Not(BeEmpty()))
+			g.Expect(backendStatus.HostnameGenerators).Should(ConsistOf(
+				hostnamegenerator_api.HostnameGeneratorStatus{
+					HostnameGeneratorRef: hostnamegenerator_api.HostnameGeneratorRef{CoreName: "static"},
+					Conditions: []hostnamegenerator_api.Condition{{
+						Type:   hostnamegenerator_api.GeneratedCondition,
+						Status: kube_meta.ConditionTrue,
+						Reason: hostnamegenerator_api.GeneratedReason,
+					}},
+				},
+			))
+		}, "2s", "100ms").Should(Succeed())
+
+		// when
+		Expect(resManager.DeleteAll(context.Background(), &meshservice_api.MeshServiceResourceList{}, core_store.DeleteAllByMesh(model.DefaultMesh))).To(Succeed())
+		Expect(resManager.DeleteAll(context.Background(), &hostnamegenerator_api.HostnameGeneratorResourceList{})).To(Succeed())
```

Comment:
Everything from this line looks like the rest of the test aside of the order of how we create other and backend.

Wouldn't this be enough to just keep
```
		Expect(other.Create(resManager, core_store.CreatedAt(createdSecond))).To(Succeed())
		Expect(backend.Create(resManager, core_store.CreatedAt(createdFirst))).To(Succeed())
```
?

This way we check that
1) we preserve the order, otherwise the test will be flaky
2) we respect the CreatedAt

Reply:
We also could keep the hostname generators in before each I think?

Reply:
> Everything from this line looks like the rest of the test aside of the order of how we create other and backend.

that's the point I think. that the order doesn't affect the outcome, even if `List` returns the resources in a different order

Reply:
> We also could keep the hostname generators in before each I think?

yeah, I added this back.